### PR TITLE
[systemtest] Add forgotten profile tags to tests

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- "Bugfix"

### Description

In some cases we didn't added tags for all stuff that we are using in our tests, this PR gonna add them to all tests. I added tags: `SCALE` -> tests where resource is scaled up/down
`ROLLING_UPDATE` -> tests where RU is triggered
`NO_ROLLING_UPDATE` -> tests where RU should not trigger
`OLM` -> tests where is OLM used for deploying the CO

Also I updated documentation (`TESTING.md`)

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=scale

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=scale

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -148,29 +148,36 @@ The following table shows currently used tags:
 
 | Name               | Description                                                                        |
 | :----------------: | :--------------------------------------------------------------------------------: |
-| travis             | Marks tests executed on Travis                                                     |
 | acceptance         | Acceptance tests, which guarantee, that basic functionality of Strimzi is working. |
 | regression         | Regression tests, which contains all non-flaky tests.                              |
 | upgrade            | Upgrade tests for specific versions of the Strimzi.                                |
+| travis             | Marks tests executed on Travis                                                     |
```
Comment (Frawless) [inline development-docs/TESTING.md:154]: You can rename it to `smoke`

File: development-docs/TESTING.md [hunk 2]
```diff
@@ -148,29 +148,36 @@ The following table shows currently used tags:
...
 | flaky              | Execute all flaky tests (tests, which are failing from time to time)               |
+| scalability        | Execute scalability tests                                                          |
```
Comment (Frawless) [inline development-docs/TESTING.md:156]: WHat's difference between this and `scale`?

File: development-docs/TESTING.md [hunk 3]
```diff
@@ -148,29 +148,36 @@ The following table shows currently used tags:
...
 | conneccomponents   | Execute tests which deploy KafkaConnect, KafkaConnectS2I, KafkaMirrorMaker2, KafkaConnector resources |
+| bridge             | Execute tests which use Kafka Bridge                                               |
 | internalclients    | Execute tests which use internal (from pod) kafka clients in tests                 |
 | externalclients    | Execute tests which use external (from code) kafka clients in tests                |
+| olm                | Execute tests which use OLM for deploy cluster operator                            |
+| metrics            | Execute tests where metrics are used                                               |
+| cruisecontrol      | Execute tests which deploy CruiseControl resource                                  |
+| scale              | Execute tests where are resources (its pods) scaled up/down                        |
+| rollingupdate      | Execute tests where is rolling update triggered                                    |
+| norollingupdate    | Execute tests where resource (and its pods) should not roll                        |
```
Comment (Frawless) [inline development-docs/TESTING.md:180]: Well....you should use only one tag and put it to every test which tests rolling update somehow from my POV.

File: development-docs/TESTING.md [hunk 4]
```diff
@@ -148,29 +148,36 @@ The following table shows currently used tags:
...
 | connectoroperator  | Execute tests which deploy KafkaConnector resource                                 |
 | connect            | Execute tests which deploy KafkaConnect resource                                   |
 | connects2i         | Execute tests which deploy KafkaConnectS2I resource                                |
 | mirrormaker        | Execute tests which deploy KafkaMirrorMaker resource                               |
 | mirrormaker2       | Execute tests which deploy KafkaMirrorMaker2 resource                              |
 | conneccomponents   | Execute tests which deploy KafkaConnect, KafkaConnectS2I, KafkaMirrorMaker2, KafkaConnector resources |
+| bridge             | Execute tests which use Kafka Bridge                                               |
 | internalclients    | Execute tests which use internal (from pod) kafka clients in tests                 |
 | externalclients    | Execute tests which use external (from code) kafka clients in tests                |
+| olm                | Execute tests which use OLM for deploy cluster operator                            |
```
Comment (Frawless) [inline development-docs/TESTING.md:175]: This is actually not true. We can execute mostly all of the tests with CO installed via OLM. This tag should be for tests, which tests examples from Strimzi manifests.

File: development-docs/TESTING.md (hunk 2 above)
Comment (im-konge) [reply development-docs/TESTING.md:156]: I was thinking about it, I split it to `scalability` -> test where big amount of users/topics are created, `scale` -> resource is scaled up/down.

But my first idea was to keep only `scalability`. Do you want to remove the `scale` tag?

File: development-docs/TESTING.md (hunk 3 above)
Comment (im-konge) [reply development-docs/TESTING.md:180]: Yep that's sound fair. I was thinking that if you use the `norollingupdate` tag when someone add some feature and you want to try, if the tests where rolling update should not be triggered are passing, you use this one to skip all other rolling update tests.

File: development-docs/TESTING.md (hunk 4 above)
Comment (im-konge) [reply development-docs/TESTING.md:175]: Thanks for correction :)

File: development-docs/TESTING.md (hunk 2 above)
Comment (michalxo) [reply development-docs/TESTING.md:156]: > I was thinking about it, I split it to `scalability` -> test where big amount of users/topics are created, `scale` -> resource is scaled up/down.
> 
> But my first idea was to keep only `scalability`. Do you want to remove the `scale` tag?

I would say, that both your cases fall into `scalability` tag group :)

Comment (im-konge) [reply development-docs/TESTING.md:156]: Okay :D I wasn't sure, thanks

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=smoke

File: systemtest/src/main/java/io/strimzi/systemtest/Constants.java [hunk 5]
```diff
@@ -129,7 +129,7 @@
     /**
      * Tag for acceptance tests executed during Travis builds.
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/Constants.java:130]: You should change also a comment.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 6]
```diff
@@ -337,6 +338,7 @@ void testJvmAndResources() {
     }
 
     @Test
+    @Tag(SCALABILITY)
     void testKafkaConnectScaleUpScaleDown() {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:342]: I am not sure if those tests should have `SCALABILITY` tag. I will leave the decision to other.

**Review by scholzj — COMMENTED** [review]
I do not think we need `ROLLING_UPDATE` and `NO_ ROLLING_UPDATE`. I think that should be in one test since those seem to be ralated to the same parts of code.

**Comment by im-konge** [comment]
@scholzj I removed the `NO_ROLLING_UPDATE` already and about the `ROLLING_UPDATE` - we have rolling update tests separated to 3 suites so I think at least this tag we should keep.

But if you don't agree, I'll remove it.

**Comment by scholzj** [comment]
Sorry, maybe it was not clear ... I'm fine to keep `ROLLING_UPDATE` ... I just thing it should be used also for the `NO_ROLLING_UPDATE` tests.

**Comment by im-konge** [comment]
@scholzj sorry, it was my misunderstanding - I already used the `ROLLING_UPDATE` tag where the rolling update should not be triggered - as you can see for dynamic logging configuration tests.

**Review by scholzj — APPROVED** [review]

**Comment by im-konge** [comment]
@scholzj I would wait, there are two nits from @Frawless to be fixed (one from them is question for everyone). Also I think that they should review this to see new tags and add their comments.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java (hunk 6 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:342]: I would keep it there, although test scales up just from 1 -> 2 pods. 
Maybe scale to 3~5 and tag is deserved there :)

**Review by michalxo — COMMENTED** [review]
Minor change in doc, else good.

File: development-docs/TESTING.md [hunk 7]
```diff
@@ -148,29 +148,34 @@ The following table shows currently used tags:
 
 | Name               | Description                                                                        |
 | :----------------: | :--------------------------------------------------------------------------------: |
-| travis             | Marks tests executed on Travis                                                     |
 | acceptance         | Acceptance tests, which guarantee, that basic functionality of Strimzi is working. |
 | regression         | Regression tests, which contains all non-flaky tests.                              |
 | upgrade            | Upgrade tests for specific versions of the Strimzi.                                |
+| smoke              | Marks tests executed on Travis                                                     |
```
[the remark this answers is not in the document]
Comment (Frawless) [reply development-docs/TESTING.md:154]: Actually it's not tided to Travis anymore.

[the remark this answers is not in the document]
Comment (im-konge) [reply development-docs/TESTING.md:154]: `Execute all smoke tests`?

**Review by michalxo — APPROVED** [review]

**Comment by michalxo** [comment]
LGTM

**Review by see-quick — APPROVED** [review]

**Comment by scholzj** [comment]
@im-konge Could you look at the conflict please?

**Comment by im-konge** [comment]
@scholzj Yep I'm already looking at it

**Comment by im-konge** [comment]
@scholzj all conflicts resolved, build is locally successful, so wait until Travis and Azure will be green and we are ready to merge

**Pull request merged**